arelle/plugin/validate/HMRC/ValidateHmrc.py

```diff
@@ -331,13 +347,54 @@ def _lang(self) -> HmrcLang:
                         return HmrcLang.WELSH
         return HmrcLang.ENGLISH
 
+    def _verifyAuditFacts(self, prefix: str) -> None:
```

Comment:
This probably wasn't clear but my intent for `_evaluateCode` was that codes would be passed in to that and the logic within it would determine how the code was evaluated. The benefits are that (a) `_evaluateCode` caches the the result so it won't be evaluated twice should that scenario ever occur, and (b) it separates the logic of the rule evaluation from actually triggering an error/warning.

Take this, for example:
https://github.com/Arelle/Arelle/blob/094f2b0c965bcabb7f27f01f3358e7492dc482e3/arelle/plugin/validate/HMRC/ValidateHmrc.py#L537-L542
This only trigger errors if the evaluation of two different codes has an unsuccessful result. If we were triggering errors during unsuccessful evaluations we would have unwanted errors in this case. It may not come up, but imagine some situation where *either* `_verifyAuditFacts` OR `_verifySomethingElse` passing should result in no errors.

So ultimately my suggestion would be to add a step to `_evaluateCode` that checks these conditions and returns an unsuccessful `CodeResult` if it fails (you can add fields to `CodeResult` to hold whatever information you need to pass into the error). Then the calls below might look like this:
```
result = self._evaluateCode(LP_AUDIT)
if not result.success:
	self._errorOnMissingAuditFacts(LP_AUDIT, result)
```

I'd agree that this is more repetitive in some ways, so I'm open to not being strict about that pattern if others feel that way.